### PR TITLE
86cueuydg/api setup

### PR DESCRIPTION
**Pull request by abhijith-biju**
## Overview
This pull request introduces creation of an Axios instance with request and response interceptors. Additionally, helper functions are implemented to facilitate the management of data in local storage.

### 1. Axios Instance Setup
- Created an Axios instance for making HTTP requests with consistent configurations.
- Implemented request interceptors to handle common headers and authentication tokens.
- Set up response interceptors for globally managing errors and responses.

### 2. Local Storage Helper Functions
- Implemented functions for saving, retrieving and deleting data in local storage.

### CODE REVIEW

File: src/api/axios.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import axios, { InternalAxiosRequestConfig, AxiosResponse } from 'axios';
+import { onRequest, onRequestError } from './requestInterceptor';
+import { onResponse, onResponseError } from './responseInterceptor';
+
+const apiClient = axios.create({
+    baseURL: import.meta.env.VITE_API_ENDPOINT as string,
+    timeout: 8000,
```
Comment (RJR-003) [inline src/api/axios.ts:7]: Could have included the timeout in variable.

File: src/utils/storageUtils.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const storage = {
+    // returns the keyValue if it exists else returns null
+    getItem: <TTokenValue>(keyName: string) => {
+        const tokenVal = localStorage.getItem(keyName)
+            ? (JSON.parse(
```
Comment (RJR-003) [inline src/utils/storageUtils.ts:5]: Can't you write this without "as string". Is this necessary?

File: src/utils/storageUtils.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                  localStorage.getItem(keyName) as string
```
Comment (RJR-003) [inline src/utils/storageUtils.ts]: Can't you write this without "as string". Is this necessary?

**Review by RJR-003 — COMMENTED** [review]
Please resolve the comments.

File: src/api/axios.ts (hunk 1 above)
Comment (ashwin8448) [inline src/api/axios.ts]: Isn't 8000ms too short?

File: src/utils/storageUtils.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+              ) as TTokenValue)
+            : null;
+
+        return tokenVal;
+    },
+
+    setItem: (keyName: string, keyValue: unknown) => {
+        window.localStorage.setItem(keyName, JSON.stringify(keyValue));
```
Comment (ashwin8448) [inline src/utils/storageUtils.ts]: Remove "window."

**Review by ashwin8448 — CHANGES_REQUESTED** [review]

File: src/utils/storageUtils.ts (hunk 4 above)
Comment (abhijith-biju) [reply src/utils/storageUtils.ts]: Removed.

File: src/utils/storageUtils.ts (hunk 3 above)
Comment (abhijith-biju) [reply src/utils/storageUtils.ts]: Removed the type assertion.

File: src/api/axios.ts (hunk 1 above)
Comment (abhijith-biju) [reply src/api/axios.ts]: changed to 30000ms.

**Review by RJR-003 — APPROVED** [review]
Reviewed.

**Review by ashwin8448 — APPROVED** [review]

File: src/api/axios.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import axios, { InternalAxiosRequestConfig, AxiosResponse } from 'axios';
+import { onRequest, onRequestError } from './requestInterceptor';
+import { onResponse, onResponseError } from './responseInterceptor';
+
+const apiClient = axios.create({
+    baseURL: import.meta.env.VITE_API_ENDPOINT as string,
```
Comment (vipinmsqburst) [inline src/api/axios.ts:6]: Is there any reason for giving name for  API base URL as VITE_API_ENDPOINT ?

Comment (abhijith-biju) [reply src/api/axios.ts:6]: I have done that to indicate it's a Vite specific environment variable.

**Review by vipinmsqburst — APPROVED** [review]

**Pull request merged**